src/v2/controllers/FormController.js

```diff
@@ -248,6 +249,16 @@ export default Controller.extend({
     let errorObj;
     if (IonResponseHelper.isIonErrorResponse(error)) {
       errorObj = IonResponseHelper.convertFormErrors(error);
+
+      // When error happened, we need to try to refresh `appState` due to `stateHandle` changed.
+      // A little sloppy logic to determine how to determine whether need to refresh appState.
+      // 1. set `hasFormError` to avoid re-render the page
+      // 2. sometime error response is very different (especially no remediation) which is only
```

Comment:
this doesn't feel right where we are updating `appState` selectively in error cases.. 

**Correct flow here--**
Always update the AppState if there was error response, But widget can choose to render or re-render
If updating AppState in error cases fails testcafe then that is a flaw in that flow. We should fix that flow instead allowing the flow to trigger error without updating the app state.. _Widget should just blindly update AppSate after every idx response._

For eg -- `pause polling when encounter 429 too many request` flow show terminal errors as form error.. Which is ideally not right. If they want the form error then the flow's remediation should dictate that...